pkg/api/converterfromapi.go

```diff
@@ -383,6 +383,17 @@ func convertOrchestratorProfileToVLabs(api *OrchestratorProfile, o *vlabs.Orches
 	if api.OrchestratorVersion != "" {
 		o.OrchestratorVersion = vlabs.OrchestratorVersion(api.OrchestratorVersion)
 	}
+
+	if api.KubernetesConfig != nil {
+		o.KubernetesConfig = &vlabs.KubernetesConfig{}
+		convertKubernetesConfigToVLabs(api.KubernetesConfig, o.KubernetesConfig)
```

Comment:
I think when customer KubernetesConfig is nil, we still insert an empty KubernetesConfig struct during the initial parsing phase. This if check is just being defensive, which I think is the right thing to do.